vignettes/formats_precedence.Rmd

```diff
@@ -0,0 +1,300 @@
+The packages used in this vignette are:
+
+```{r, message=FALSE}
+library(rtables)
+```
+
+The examples shown in this vignette are based on the example `ADSL` dataset, i.e. a demographic table that summarizes the variables content for different population subsets (encoded in the columns).
+
+```{r}
+ADSL <- ex_adsl
+```
+
+Note that all `ex_*` data which is currently attached to the `rtables` package is provided by the [`formatters`](https://insightsengineering.github.io/formatters/) package and was created using the publicly available [`random.cdisc.data`](https://insightsengineering.github.io/random.cdisc.data/) R package.
+
+### Formats precedence and inheritance rules
+
+The format in which numbers are printed can be specified by the user in a few different places. In the context of the precedence, it is important at which level of the split hierarchy formats are specified. In general, there are two such levels: the **cell** level or the so-called **parent table** level. The concept of the cell and the parent table result from the way in which `rtables` package stores resulting tables. It models the resulting tables as hierarchical, tree-like objects with the cells (as leaves) containing multiple values. Particularly noteworthy in this context is the fact that the actual table splitting occurs in a row-dominant way (even if the column split is present in the layout). `rtables` provides user-end function `table_structure()` that prints the structure of a given table object. For a quick illustration, consider the following example.
+
+```{r}
+lyt <- basic_table() %>%
+  split_cols_by("ARM") %>%
+  split_rows_by("SEX") %>%
+  analyze(vars = "AGE", afun = mean)
+
+adsl_analyzed <- build_table(lyt, ADSL)
+adsl_analyzed
+table_structure(adsl_analyzed)
+```
+
+For this reporting table, there are 4 sub-tables under `SEX` table, i.e. `F`, `M`, `U`, and `UNDIFFERENTIATED`. Each of these sub-tables has one sub-table `AGE`. So, e.g. for the top sub-table `AGE`, its nearest parent table is `F`.
+
+The concept of formats precedence and inheritance in `rtables` can have multiple levels of parent tables and a final cell level. The cell level is the most specific and lowest level at which the format can be specified by the user. If the cell format is not explicitly specified, it is then inherited from its nearest parent table for which the format was chosen by the user (if any). If no specific format has been selected by the user, then the default format will be applied. The default format is "xx" and it yields the same formatting as `as.character()` function. In summary, the precedence-inheritance chain looks like below.
+```
+cell <- parent_table <- parent_table <- ... <- parent_table
+```
+In case when the format is specified by the user in more than one place, the one which is closer to the `cell` in the precedence-inheritance chain will be applied. In the following sections of this vignette, we illustrate the format precedence rules with a few examples.
```

Comment:
Please use "most specific" rather than "closest to cell" when describing which will be apply to a particular value.